src/server/ApiRoutes/Hold.js

```diff
@@ -54,24 +68,84 @@ function postHoldAPI(req, pickedUpItemId, cb, errorCb) {
     .catch(errorCb);
 }
 
+/**
+ * getDeliveryLocations(barcode, patronId, accessToken, cb, errorCb)
+ * The function to make a request to get delivery locations of an item.
+ *
+ * @param {string} barcode
+ * @param {string} patronId
+ * @param {string} accessToken
+ * @param {function} cb - callback when we have valid response
+ * @param {function} errorCb - callback when error
+ * @return {function}
+ */
+function getDeliveryLocations(barcode, patronId, accessToken, cb, errorCb) {
+  return axios.get(
+    `${apiBase}/request/deliverylocationsbybarcode?barcodes[]=${barcode}&patronId=${patronId}`,
+    {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  )
+  .then(barcodeAPIresponse => {
+    cb(
+      barcodeAPIresponse.data.itemListElement[0].deliveryLocation,
+      barcodeAPIresponse.data.itemListElement[0].eddRequestable
+    );
+  })
+  .catch(barcodeAPIError => {
+    errorCb(barcodeAPIError);
+  });
+}
+
+/**
+ * confirmRequestServer(req, res, next)
+ * The function to return the bib and item data with its delivery locations to confirmation page.
+ *
+ * @param {req}
+ * @param {res}
+ * @param {next}
+ * @return {function}
+ */
 function confirmRequestServer(req, res, next) {
   const bibId = req.params.bibId || '';
   const loggedIn = User.requireUser(req, res);
+  const error = req.query.error ? JSON.parse(req.query.error) : {};
 
   if (!loggedIn) return false;
 
+  const accessToken = req.tokenResponse.accessToken || '';
+  const patronId = req.tokenResponse.decodedPatron.sub || '';
+  let barcode;
+
+  // Retrieve item
   return Bib.fetchBib(
     bibId,
-    (data) => {
-      res.locals.data.Store = {
-        bib: data,
-        searchKeywords: '',
-        error: {},
-      };
-      next();
+    (bibResponseData) => {
+      barcode = LibraryItem.getItem(bibResponseData, req.params.itemId).barcode;
+
+      getDeliveryLocations(
+        barcode,
+        patronId,
+        accessToken,
+        (deliveryLocations, isEddRequestable) => {
+          res.locals.data.Store = {
+            bib: bibResponseData,
+            searchKeywords: '',
+            error,
+            deliveryLocations,
+            isEddRequestable,
+          };
+          next();
+        },
+        (e) => {
+          console.error(`deliverylocationsbybarcode API error: ${JSON.stringify(e, null, 2)}`);
```

Comment:
If there's an error, you want to do what the outer function is doing, set up the Store with the error and call `next()` (like in line 148).